index.js

```diff
@@ -7,6 +7,43 @@ const PLAYING = 2;
 const MODE_LERP = 0;
 const MODE_HERMITE = 1;
 
+const vectorPool = [];
+const quatPool = [];
+const framePool = [];
+
+const getPooledVector = () => vectorPool.shift() || new THREE.Vector3();
```

Comment:
I *think* `pop` performs better than `shift` though I have had a hard time finding evidence of this. Here are two places where this is discussed but they failed to convince me. (Neither really addressed the fact that it depends on the implementation in the js engine.)

http://taoalpha.github.io/blog/2015/10/30/tech-archived-javascript-array-operation-cost/

https://stackoverflow.com/questions/6501160/why-is-pop-faster-than-shift

Reply:
`pop()` seems smarter anyway